app/components/primer/component.rb

```diff
@@ -6,7 +6,7 @@
 module Primer
   # @private
   class Component < ViewComponent::Base
-    include ViewComponent::SlotableV2 unless ViewComponent::VERSION::STRING.to_f >= 2.28
+    include ViewComponent::SlotableV2 unless ViewComponent::VERSION::MINOR >= 28
```

Comment:
what if we use `Gem::Version`?
`Gem::Version.new(ViewComponent::VERSION::STRING) >= Gem::Version.new("2.28.0")`
it is more verbose, but we wouldn't need to update this when we support v3

Reply:
```suggestion
    include ViewComponent::SlotableV2 unless ViewComponent::Base < ViewComponent::SlotableV2
```

Would this work?

Reply:
I have **no idea** of what that does 😅

Reply:
https://ruby-doc.org/core-2.7.3/Module.html#3C-method is the method docs.

It returns `true` if `ViewComponent::Base` inherits from `ViewComponent::SlotableV2` in some way (`include` modifies the inheritance chain).

```ruby
>> ViewComponent::Base < ViewComponent::SlotableV2
=> true
>> ViewComponent::Base.ancestors
=> [ViewComponent::Base, ViewComponent::WithContentHelper, ViewComponent::SlotableV2, ViewComponent::Previewable, ActiveSupport::Configurable, Haml::Helpers::ActionViewMods, ActionView::Base, Sprockets::Rails::Helper, Sprockets::Rails::Utils, ActionView::Helpers, ActionView::Helpers::RenderingHelper, ActionView::Helpers::NumberHelper, ActionView::Helpers::JavaScriptHelper, ActionView::Helpers::FormOptionsHelper, ActionView::Helpers::TranslationHelper, ActionView::Helpers::FormHelper, ActionView::RecordIdentifier, ActionView::ModelNaming, ActionView::Helpers::FormTagHelper, ActionView::Helpers::TextHelper, ActionView::Helpers::DebugHelper, ActionView::Helpers::DateHelper, ActionView::Helpers::CsrfHelper, ActionView::Helpers::CspHelper, ActionView::Helpers::ControllerHelper, ActionView::Helpers::CacheHelper, ActionView::Helpers::AtomFeedHelper, ActionView::Helpers::AssetTagHelper, ActionView::Helpers::SanitizeHelper, ActionView::Helpers::AssetUrlHelper, ActionView::Helpers::ActiveModelHelper, ActiveSupport::Benchmarkable, ActionView::Helpers::UrlHelper, ActionView::Helpers::TagHelper, ActionView::Helpers::OutputSafetyHelper, ActionView::Helpers::CaptureHelper, ERB::Util, ActionView::Context, ActiveSupport::Dependencies::ZeitwerkIntegration::RequireDependency, ActiveSupport::ToJsonWithActiveSupportEncoder, Object, ActiveSupport::Dependencies::Loadable, ActiveSupport::Tryable, Minitest::Expectations, PP::ObjectMixin, JSON::Ext::Generator::GeneratorMethods::Object, Kernel, BasicObject]
```

Reply:
🤯

Reply:
https://ruby-doc.org/core-2.7.3/Module.html#method-i-include-3F would also probably work here